File: PHP_native/test_insert.php
<?php
$host = '127.0.0.1';
$db   = 'bp_srovnani';
$user = 'root';
$pass = '';

$dsn = "mysql:host=$host;dbname=$db;charset=utf8mb4";
$pdo = new PDO($dsn, $user, $pass);

$iterace = 100;
$casy = [];

for ($i = 0; $i < $iterace; $i++) {
    $start_time = microtime(true);
    
    // V každé iteraci vložíme 10 záznamů
    for ($j = 0; $j < 10; $j++) {
        $text = "Benchmark test zprávy " . rand(1, 10000);
        // Předpokládáme, že uživatel s ID 1 a kategorie s ID 1 v DB existují (z předchozího plnění)
        $sql = "INSERT INTO zpravy (id_uzivatele, id_kategorie, text_zpravy) VALUES (1, 1, '$text')";
        $pdo->exec($sql);
    }
    
    $end_time = microtime(true);
    $casy[] = ($end_time - $start_time) * 1000;
}

$prumerny_cas = array_sum($casy) / count($casy);

header('Content-Type: application/json');
echo json_encode([
    'jazyk' => 'PHP Nativní',
    'test' => 'Hromadný zápis (INSERT 10x na iteraci)',
    'pocet_iteraci' => $iterace,
    'celkem_zapsano_radku' => $iterace * 10,
    'prumerny_cas_ms' => round($prumerny_cas, 2)
]);
?>

File: Python-flask/test_insert.py
import time
import random
from flask import Flask, jsonify
import mysql.connector

app = Flask(__name__)

@app.route('/insert')
def test_insert():
    iterace = 100
    casy = []
    
    db_config = {
        'host': '127.0.0.1',
        'user': 'root',
        'password': '',
        'database': 'bp_srovnani'
    }
    
    conn = mysql.connector.connect(**db_config)

    for _ in range(iterace):
        start_time = time.time()
        
        cursor = conn.cursor()
        # V každé iteraci vložíme 10 záznamů
        for _ in range(10):
            text = f"Benchmark test zprávy {random.randint(1, 10000)}"
            sql = f"INSERT INTO zpravy (id_uzivatele, id_kategorie, text_zpravy) VALUES (1, 1, '{text}')"
            cursor.execute(sql)
            
        conn.commit() # Uložení změn do databáze
        cursor.close()
        
        end_time = time.time()
        casy.append((end_time - start_time) * 1000)
        
    conn.close()
        
    prumerny_cas = sum(casy) / len(casy)
    
    return jsonify({
        'jazyk': 'Python Flask',
        'test': 'Hromadný zápis (INSERT 10x na iteraci)',
        'pocet_iteraci': iterace,
        'celkem_zapsano_radku': iterace * 10,
        'prumerny_cas_ms': round(prumerny_cas, 2)
    })

if __name__ == '__main__':
    app.run(port=5000)